pkg/azdiskschedulerextender/azdiskschedulerextender_test.go

```diff
@@ -840,14 +937,42 @@ func cleanConfigAndRestoreEnv(path string, envVariableName string, envValue stri
 	os.Remove(path)
 }
 
-func setupTestInformers(kubeExtensionClientset versionedClientSet.Interface) {
-	informerFactory := informers.NewSharedInformerFactory(kubeExtensionClientset, noResyncPeriodFunc())
-	azVolumeAttachmentInformer = informerFactory.Disk().V1alpha2().AzVolumeAttachments()
-	azDriverNodeInformer = informerFactory.Disk().V1alpha2().AzDriverNodes()
+func setupTestInformers(versionedInterface versionedClientSet.Interface, coreInterface kubernetes.Interface) {
+	v1alpah1InformerFactory := azurediskInformers.NewSharedInformerFactory(kubeExtensionClientset, noResyncPeriodFunc())
+	coreInformerFactory := informers.NewSharedInformerFactory(coreInterface, noResyncPeriodFunc())
+
+	azVolumeAttachmentInformer = v1alpah1InformerFactory.Disk().V1alpha2().AzVolumeAttachments()
+	azDriverNodeInformer = v1alpah1InformerFactory.Disk().V1alpha2().AzDriverNodes()
+	pvInformer = coreInformerFactory.Core().V1().PersistentVolumes()
+	pvcInformer = coreInformerFactory.Core().V1().PersistentVolumeClaims()
 
 	stopper := make(chan struct{})
 	defer close(stopper)
 	go azVolumeAttachmentInformer.Informer().Run(stopper)
 	go azDriverNodeInformer.Informer().Run(stopper)
-	cache.WaitForCacheSync(stopper, azVolumeAttachmentInformer.Informer().HasSynced, azDriverNodeInformer.Informer().HasSynced)
+	go pvInformer.Informer().Run(stopper)
+	go pvcInformer.Informer().Run(stopper)
+	cache.WaitForCacheSync(stopper, azVolumeAttachmentInformer.Informer().HasSynced, azDriverNodeInformer.Informer().HasSynced, pvInformer.Informer().HasSynced, pvcInformer.Informer().HasSynced)
+	recover()
+}
+
+func recover() error {
```

Comment:
np: Since this is now used only in the context of unittest, `recover` doesn't communicate the intend. Maybe we should rename the function `populateCache`.